bazarr.py

```diff
@@ -49,55 +52,77 @@ def get_python_path():
+    ep = subprocess.Popen(script, stdout=None, stderr=None, stdin=subprocess.DEVNULL)
+    print(f"Bazarr starting child process with PID {ep.pid}...")
+    return ep
+ 
+
+def terminate_child():
+    print(f"Terminating child process with PID {child_process.pid}")
+    child_process.terminate()
 
-def end_child_process(ep):
+
+def get_stop_status_code(input_file):
     try:
-        if os.name != 'nt':
-            try:
-                ep.send_signal(signal.SIGINT)
-            except ProcessLookupError:
-                pass
-        else:
-            import win32api
-            import win32con
+        with open(input_file,'r') as file:
+            # read status code from file, if it exists
+            line = file.readline()
             try:
-                win32api.GenerateConsoleCtrlEvent(win32con.CTRL_C_EVENT, ep.pid)
-            except KeyboardInterrupt:
-                pass
+                status_code = int(line)
+            except (ValueError, TypeError):
+                status_code = EXIT_NORMAL
+            file.close()
     except:
-        ep.terminate()
-
-
-def start_bazarr():
-    script = [get_python_path(), "-u", os.path.normcase(os.path.join(dir_name, 'bazarr', 'main.py'))] + sys.argv[1:]
-    ep = subprocess.Popen(script, stdout=None, stderr=None, stdin=subprocess.DEVNULL)
-    atexit.register(end_child_process, ep=ep)
-    signal.signal(signal.SIGTERM, lambda signal_no, frame: end_child_process(ep))
+        status_code = EXIT_NORMAL
+    return status_code
 
 
 def check_status():
+    global child_process
     if os.path.exists(stopfile):
+        status_code = get_stop_status_code(stopfile)
         try:
+            print(f"Deleting stop file...")
             os.remove(stopfile)
-        except Exception:
+        except Exception as e:
             print('Unable to delete stop file.')
         finally:
-            print('Bazarr exited.')
-            sys.exit(0)
+            terminate_child()
+            exit_program(status_code)
 
     if os.path.exists(restartfile):
         try:
+            print(f"Deleting restart file...")
             os.remove(restartfile)
         except Exception:
             print('Unable to delete restart file.')
-        else:
-            print("Bazarr is restarting...")
-            start_bazarr()
+        finally:
+            terminate_child()
+            print(f"Bazarr is restarting...")
+            child_process = start_bazarr()
+
+
+def interrupt_handler(signum, frame):
+    # catch and ignore keyboard interrupt Ctrl-C
+    # the child process Server object will catch SIGINT and perform an orderly shutdown
+    global interrupted
+    if not interrupted:
+        # ignore user hammering Ctrl-C; we heard you the first time!
+        interrupted = True
+        print('Handling keyboard interrupt...')
+    else:
+        print(f"Stop doing that! I heard you the first time!")
```

Comment:
Nice one ;-)

Reply:
It actually does help if the user does that.
Otherwise, it would “restart” the shutdown process.